src/SFA.DAS.ApprenticeCommitments.Web/Pages/ConfirmYourIdentity.cshtml.cs

```diff
@@ -60,8 +61,17 @@ public class ConfirmYourIdentityModel : PageModel
             return Page();
         }
 
-        public IActionResult OnPost()
+        public async Task<IActionResult> OnPost()
         {
+            Guid.TryParse(User.Claims.FirstOrDefault(c => c.Type == "sub")?.Value ?? "", out var regId);
```

Comment:
yes it does need to be abstracted away, perhaps by an extention method or a user factory

Reply:
Definitely, that's #11 :)